src/webhooks/handleCheckoutSessionCompleted.ts

```diff
@@ -59,12 +59,25 @@ export default async function handleCheckoutSessionCompleted(
     throw err;
   }
 
+  try {
+    const userData = await usersService.findUserByUuid(user.uuid);
+    const invoice = await stripe.invoices.retrieve(session.invoice as string);
+
+    const couponId = invoice.discount?.coupon.id;
+
+    if (couponId) {
+      await usersService.storeCouponUsedByUser(userData, couponId);
+    }
+  } catch (err) {
+    log.error(`Error while adding user ${user.uuid} and coupon`);
+
+    throw err;
```

Comment:
Do not propagate, the `storeCouponUsedByUser` in fact throws an error if the coupon is not being tracked, but this could be intentional. In fact, you may log the error only if the error is not `instanceof` that error the function throws